### PR TITLE
FIX-#4927: Fix `dtypes` computation in `dataframe.filter`

### LINKED ISSUE #4927: BUG: `dtypes` wrong computation in `dataframe.filter`
Place: https://github.com/modin-project/modin/blob/19ac8ff208eb02a7f86037e338ce306f1e96d7a9/modin/core/dataframe/pandas/dataframe/dataframe.py#L1882

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4927 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/dataframe/pandas/dataframe/dataframe.py [hunk 1]
```diff
@@ -1879,7 +1879,7 @@ def filter(self, axis: Union[Axis, int], condition: Callable) -> "PandasDatafram
             new_partitions,
             *new_axes,
             *new_lengths,
-            self.dtypes if axis == 0 else None,
+            self.dtypes if axis.value == 1 else None,
```
Comment (YarShev) [inline modin/core/dataframe/pandas/dataframe/dataframe.py:1882]:
```suggestion
            self.dtypes if axis == Axis.ROW_WISE else None,
```

**Review by pyrito — COMMENTED** [review]
Would it be possible to have a test for this at all? Could we check to see that the `dtypes` are not None or something?

File: modin/core/dataframe/pandas/dataframe/dataframe.py (hunk 1 above)
Comment (anmyachev) [reply modin/core/dataframe/pandas/dataframe/dataframe.py:1882]: In that case it's needed to use `Axis.COL_WISE`.

**Comment by anmyachev** [comment]
> Would it be possible to have a test for this at all? Could we check to see that the `dtypes` are not None or something?

I will add.

**Review by pyrito — COMMENTED** [review]
Thanks for adding the test! Looks like there is a CI failure

File: modin/core/dataframe/pandas/dataframe/dataframe.py (hunk 1 above)
Comment (YarShev) [reply modin/core/dataframe/pandas/dataframe/dataframe.py]: right

**Review by pyrito — APPROVED** [review]

**Review by YarShev — APPROVED** [review]

**Pull request merged**